File: eBarbershop.Model/Termin.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eBarbershop.Model
{
    public class Termin
    {
        public int TerminId { get; set; }
        [DisplayName("Termin kod")]
        public string TerminUposelnik => $"{Korisnik?.Ime} {Korisnik?.Prezime}";
        public DateTime Vrijeme { get; set; }

        public int RezervacijaId { get; set; }
        public Rezervacija Rezervacija { get; set; }
        public bool isBooked { get; set; }

        [Browsable(false)]
        public int KorisnikID { get; set; }
        [Browsable(false)]
        public Korisnik Korisnik { get; set; }

    }

}

File: UI/ebarbershop_admin/lib/models/termin.dart
import 'package:ebarbershop_admin/models/korisnik.dart';
import 'package:ebarbershop_admin/models/rezervacija.dart';
import 'package:json_annotation/json_annotation.dart';

part 'termin.g.dart';

@JsonSerializable()
class Termin {
  int? terminId;
  String? terminUposelnik;
  DateTime? vrijeme;
  int? rezervacijaId;
  Rezervacija? rezervacija;
  bool? isBooked;
  int? korisnikID;
  Korisnik? korisnik;

  Termin(
    this.terminId,
    this.terminUposelnik,
    this.vrijeme,
    this.rezervacijaId,
    this.rezervacija,
    this.isBooked,
    this.korisnikID,
    this.korisnik,
  );

  factory Termin.fromJson(Map<String, dynamic> json) => _$TerminFromJson(json);
  Map<String, dynamic> toJson() => _$TerminToJson(this);
}
